LibertyDSNP/parquetjs
# Parquetjs browser support

# Problem 
Need to be able to use parquetjs in browser, [Fixes #178456668](https://www.pivotaltracker.com/story/show/178456668)

with @enddynayn , @acruikshank , @aneyzberg 

# Solution Summary
1. Configuration for bundling parquetjs that works in browser, via esbuild
1. Add to build targets
1. Fix all the things that got broken 
1. Example server showing how it can be used
1. Updates to README
1. Disable LZO and Brotli in browser (see notes)
1. Disable LZO in Node as a consequence of this attempt (see notes)

# To verify:
1. Build parquetjs
1. Make sure tests all pass
1. Use the esbuild instructions to serve up the built parquetjs
1. Fire up the example server as shown in its README and play with it. Be sure to open your console.  The only problems you should see are with anything marked "unsupported."  Otherwise you should see the data in console.
 
Notes:
* We attempted to get Brotli compression working in  browser, almost got it, but hit a serious road bump with WebAssembly calls. There was a lot of work involved in the "almost", so since we may wish to get this working at a later date, rather than backing out of these changes and putting back the old brotli library, I have left these changes in just for NodeJS.
* Turns out LZO is was already horribly broken, so it's disabled for NodeJS too.
* Strangely, XXhash wasm library works just fine.
* I had fixed a few other small bugs while trying to get this working, but I took the changes out of this PR and will submit later.

## Example server working:
![example-server-md](https://user-images.githubusercontent.com/502640/124013163-d6fa9400-d996-11eb-95e0-32005db7bae0.gif)
